src/njoy21/input/MATXSR/src/readCard10List.hpp

```diff
@@ -0,0 +1,25 @@
+template< typename Istream >
+std::vector< Card10 > readCard10List( Istream& is, const unsigned int nmat )
+{
+  std::vector< Card10 > card10List;
+
+  for( unsigned int i = 0; i < nmat; i++ ){
+    try{
+      card10List.emplace_back( is );
+    }
+    catch( std::exception& e ){
+      Log::info( "Trouble reading list of Card 10 values." );
+      throw e;
+    }
+  }
+
+  try{
```

Comment:
Trying to read one more `Card10` here is problematic. Let's assume that it fails—as is the expected behavior. The exception is caught and we move along. However, the `Istream` iterator is now in the wrong place for the next module. We've already read the first part of it and thrown it away. 

I think what you are trying to do is identify if there are too many `Card10`s. Unfortunately, we can't really determine that here. However, whatever is read next will (should?) fail and that's where we'll catch that there are too many.